dipy/direction/ptt_direction_getter.pyx

```diff
@@ -0,0 +1,472 @@
+# cython: boundscheck=False
+# cython: initializedcheck=False
+# cython: wraparound=False
+
+"""
+Implementation of the Parallel Transport Tractography (PTT) algorithm by
+Aydogan et al., (2021).
+
+Aydogan DB, Shi Y. Parallel Transport Tractography. IEEE Trans
+    Med Imaging. 2021 Feb;40(2):635-647. doi: 10.1109/TMI.2020.3034038.
+    Epub 2021 Feb 2. PMID: 33104507; PMCID: PMC7931442.
+"""
+
+cimport numpy as cnp
+from libc.math cimport M_PI, pow, sin, cos
+
+from dipy.direction.probabilistic_direction_getter cimport \
+        ProbabilisticDirectionGetter
+from dipy.utils.fast_numpy cimport (copy_point, cross, norm, normalize, random,
+                                    random_vector, random_perpendicular_vector,
+                                    random_point_within_circle)
+from dipy.tracking.stopping_criterion cimport (StreamlineStatus,
+                                               StoppingCriterion,
+                                               TRACKPOINT,
+                                               ENDPOINT,
+                                               OUTSIDEIMAGE,
+                                               INVALIDPOINT)
+from dipy.tracking.utils import min_radius_curvature_from_angle
+
+
+cdef class PTTDirectionGetter(ProbabilisticDirectionGetter):
+    """Parallel Transport Tractography (PTT) direction getter.
+    """
+    cdef double       angular_separation
+    cdef double       data_support_exponent
+    cdef double[3][3] frame
+    cdef double       k1
+    cdef double       k2
+    cdef double       k_small
+    cdef double       last_val
+    cdef double       last_val_cand
+    cdef double       max_angle
+    cdef double       max_curvature
+    cdef double[3]    position
+    cdef int          probe_count
+    cdef double       probe_length
+    cdef double       probe_normalizer
+    cdef int          probe_quality
+    cdef double       probe_radius
+    cdef double       probe_step_size
+    cdef double[9]    propagator
+    cdef double       step_size
+
+
+    def __init__(self, pmf_gen, max_angle, sphere, pmf_threshold=None,
+                 double probe_length=0.5, double probe_radius=0,
+                 int probe_quality=3, int probe_count=1,
+                 double data_support_exponent=1, **kwargs):
```

Comment:
Because the default values are slightly different in Trekker, the results will not be the same, which is of course OK. But this might be good to mention in the documentation later on. (For example, probe_length's default value 0.25 x voxelSize in Trekker.)

Reply:
Can you address this @gabknight ? thank you

Reply:
Ok thanks. I clarified this in the doc.